npm install fails on docify created website

## Bug Report

### Steps to Reproduce:
 1. In calm repo
 2. `cd calm/architecture; calm docify --architecture calm-2.json --output website`
 2. `cd website`
 3. `npm i`

### Expected Result:
Successful npm install of required docusaurus packages.

### Actual Result:
```
xxxx@xxxxxx:~/architecture-as-code/calm/architecture/website$ npm i
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: @docusaurus/theme-mermaid@3.10.0
npm error Found: @mermaid-js/layout-elk@0.2.1
npm error node_modules/@mermaid-js/layout-elk
npm error   @mermaid-js/layout-elk@"^0.2.0" from the root project
npm error
npm error Could not resolve dependency:
npm error peerOptional @mermaid-js/layout-elk@"^0.1.9" from @docusaurus/theme-mermaid@3.10.0
npm error node_modules/@docusaurus/theme-mermaid
npm error   @docusaurus/theme-mermaid@"^3.9.0" from the root project
npm error
npm error Conflicting peer dependency: @mermaid-js/layout-elk@0.1.9
npm error node_modules/@mermaid-js/layout-elk
npm error   peerOptional @mermaid-js/layout-elk@"^0.1.9" from @docusaurus/theme-mermaid@3.10.0
npm error   node_modules/@docusaurus/theme-mermaid
npm error     @docusaurus/theme-mermaid@"^3.9.0" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry this command with --force or --legacy-peer-deps to accept an incorrect (and potentially broken) dependency resolution.
```

### Environment:
Ubuntu 24 (within Windows 11 WSL2)
CALM CLI 1.37.0
Node 24.14.1

### Additional Context:
Also seen by @yt-ms

--- a/cli/test_fixtures/getting-started/STEP-2/website/package.json
+++ b/cli/test_fixtures/getting-started/STEP-2/website/package.json
@@ -20,5 +20,9 @@
     "react": "^18.0.0",
     "react-dom": "^18.0.0",
     "unist-util-visit": "^5.0.0"
+  },
+  "overrides": {
+    "@mermaid-js/layout-elk": "^0.2.0",
+    "webpackbar": "^7.0.0"
   }
 }

--- a/cli/test_fixtures/getting-started/STEP-3/website/package.json
+++ b/cli/test_fixtures/getting-started/STEP-3/website/package.json
@@ -20,5 +20,9 @@
     "react": "^18.0.0",
     "react-dom": "^18.0.0",
     "unist-util-visit": "^5.0.0"
+  },
+  "overrides": {
+    "@mermaid-js/layout-elk": "^0.2.0",
+    "webpackbar": "^7.0.0"
   }
 }

--- a/shared/src/docify/template-bundles/docusaurus/package.json
+++ b/shared/src/docify/template-bundles/docusaurus/package.json
@@ -20,5 +20,9 @@
     "react": "^18.0.0",
     "react-dom": "^18.0.0",
     "unist-util-visit": "^5.0.0"
+  },
+  "overrides": {
+    "@mermaid-js/layout-elk": "^0.2.0",
+    "webpackbar": "^7.0.0"
   }
 }

--- a/shared/src/docify/template-bundles/docusaurus/package.spec.ts
+++ b/shared/src/docify/template-bundles/docusaurus/package.spec.ts
@@ -1,0 +1,34 @@
+import { describe, it, expect, afterAll } from 'vitest';
+import { copyFileSync, mkdtempSync, rmSync } from 'fs';
+import { join } from 'path';
+import { tmpdir } from 'os';
+import { spawn } from 'child_process';
+
+const TEMPLATE_PACKAGE_JSON = join(__dirname, 'package.json');
+
+// The purpose of this test is to ensure that the docusaurus template bundle's dependencies can be
+// resolved. This is a long-running test because it involves "npm install", which can take some time.
+//
+// Without this test, we might not discover peer dependency conflicts until after a release is produced
+// and a user tries to use the template bundle.
+describe('docusaurus template bundle dependencies', () => {
+    const tmpDir = mkdtempSync(join(tmpdir(), 'calm-docusaurus-'));
+
+    afterAll(() => {
+        rmSync(tmpDir, { recursive: true, force: true });
+    });
+
+    it('should resolve without peer dependency conflicts (long running test)', async () => {
+        copyFileSync(TEMPLATE_PACKAGE_JSON, join(tmpDir, 'package.json'));
+
+        const exitCode = await new Promise<number>((resolve, reject) => {
+            const stderr: string[] = [];
+            const proc = spawn('npm', ['install', '--package-lock-only', '--prefix', tmpDir]);
+            proc.stderr?.on('data', (chunk: Buffer) => stderr.push(chunk.toString()));
+            proc.on('close', (code) => resolve(code ?? 1));
+            proc.on('error', reject);
+        });
+
+        expect(exitCode).toBe(0);
+    }, 120_000);
+});

--- a/shared/test_fixtures/docify/workshop/expected-output/non-secure/package.json
+++ b/shared/test_fixtures/docify/workshop/expected-output/non-secure/package.json
@@ -20,5 +20,9 @@
     "react": "^18.0.0",
     "react-dom": "^18.0.0",
     "unist-util-visit": "^5.0.0"
+  },
+  "overrides": {
+    "@mermaid-js/layout-elk": "^0.2.0",
+    "webpackbar": "^7.0.0"
   }
 }

--- a/shared/test_fixtures/docify/workshop/expected-output/secure/package.json
+++ b/shared/test_fixtures/docify/workshop/expected-output/secure/package.json
@@ -20,5 +20,9 @@
     "react": "^18.0.0",
     "react-dom": "^18.0.0",
     "unist-util-visit": "^5.0.0"
+  },
+  "overrides": {
+    "@mermaid-js/layout-elk": "^0.2.0",
+    "webpackbar": "^7.0.0"
   }
 }

